### PR TITLE
MONGOID-4172: Use positional operator only on 1 level deep nesting.

### PR DESCRIPTION
**Pull request by durran**

### CODE REVIEW

File: lib/mongoid/positional.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    def process_updates(keys, update, updates = {})
+      update.each_pair do |position, value|
+        updates[replace_index(keys, position)] = value
+      end
+      updates
+    end
+
+    def replace_index(keys, position)
+      # replace index with $ only if that key is on the selector and it is only
```
Comment (estolfo) [inline lib/mongoid/positional.rb]: key is *in* the selector

File: spec/mongoid/positional_spec.rb [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+        let(:selector) do
+          { "_id" => 1, "children._id" => nil }
+        end
+
+        let(:expected) do
+          {
+            "$set" => {
+              "field" => "value",
+              "children.0.field" => "value",
```
Comment (estolfo) [inline spec/mongoid/positional_spec.rb]: shouldn't this be 

```ruby
"children.$.field" => "value",
```

Comment (durran) [inline spec/mongoid/positional_spec.rb]: No because the _id is nil on the second child.

Comment (estolfo) [inline spec/mongoid/positional_spec.rb]: ok so shouldn't the test say it doesn't do a replacement?

Comment (durran) [inline spec/mongoid/positional_spec.rb]: Yeah - I updated the test wording.

Comment (estolfo) [inline spec/mongoid/positional_spec.rb]: ok makes sense now = )

**Pull request merged**